enterprise-suite/es-monitor-api/prometheus.yml

```diff
@@ -148,23 +145,29 @@ scrape_configs:
 
     # workload labels:
     metric_relabel_configs:
-      - source_labels: [created_by_kind, pod]
-        regex: 'ReplicaSet;(.*)-[^-]+-[^-]+'
+      - source_labels: [created_by_kind, pod, workload]
+        regex: 'ReplicaSet;(.+)-[^-]+-[^-]+;'
         target_label: workload
-      - source_labels: [created_by_kind, pod]
-        regex: 'StatefulSet;(.*)-[^-]+'
+      - source_labels: [created_by_kind, pod, workload]
+        regex: 'StatefulSet;(.+)-[^-]+;'
         target_label: workload
-      - source_labels: [created_by_kind, pod]
-        regex: 'DaemonSet;(.*)-[^-]+'
+      - source_labels: [created_by_kind, pod, workload]
+        regex: 'DaemonSet;(.+)-[^-]+;'
         target_label: workload
-      - source_labels: [created_by_kind, pod]
-        regex: 'Job;(.*)-[^-]+'
+      - source_labels: [created_by_kind, pod, workload]
+        regex: 'Job;(.+)-[^-]+;'
         target_label: workload
-      - source_labels: [created_by_kind, pod]
-        regex: 'ReplicationController;(.*)-[^-]+'
+      - source_labels: [created_by_kind, pod, workload]
+        regex: 'ReplicationController;(.+)-[^-]+;'
         target_label: workload
-      - source_labels: [created_by_kind, pod]
-        regex: '.none.;(.*)'
+      - source_labels: [created_by_kind, pod, workload]
+        regex: '.none.;(.+);'
+        target_label: workload
+      - source_labels: [deployment, workload]
+        regex: '(.+);'
+        target_label: workload
+      - source_labels: [daemonset, workload]
+        regex: '(.+);'
```

Comment:
Don't we need these rules for the pods? How come they are only added to `Service` service discovery? Which I don't think we're using anywhere, are we?

Reply:
this section is for kube-state metrics

Reply:
hmm replied on the other view dunno why it isn't flowing here as a response as well.  github bug?!

anyway, these `created_by_kind` etc are from kube state metrics, which we scrape this way.  We could scope the job to only find kube state metrics, just haven't thought about it.

This change is about getting the relabeling we are doing to be consistent, maybe we can add other issues if we want to refactor the config more?